integration-tests/src/test/resources/deploywebapp.py

```diff
@@ -1,5 +1,22 @@
 # Copyright 2018, Oracle Corporation and/or its affiliates.  All rights reserved.
 # Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
 
+import time as systime
+
 connect(sys.argv[1],sys.argv[2],sys.argv[3])
-deploy(sys.argv[4],sys.argv[5],sys.argv[6],upload='false',remote='false')
\ No newline at end of file
+deploy(sys.argv[4],sys.argv[5],sys.argv[6],upload='false',remote='false')
+systime.sleep(30)
```

Comment:
This seems like a long sleep.   It might be too long on some systems - making the tests run too long, or too short on others - cuasing the 'state print' you added below to be invalid.   Is there a way you can have WLST loop every  second to wait until '/AppDeployments/'+appName.getName()+'/Targets' and its associated runtime exist?

Reply:
Actually I have added the app status code for my debugging. I think I should remove it. Is it needed?

Reply:
Since it adds 30 seconds, and its only debug, maybe just make it optional (default off), or leave it commented out...   Unless you can figure out how to do a loop/test of some sort...

Reply:
commented.